router/src/history/params.rs

```diff
@@ -78,7 +78,7 @@ macro_rules! params_map {
         let start_capacity = common_macros::const_expr_count!($($key);*);
         #[allow(unused_mut)]
         let mut map = linear_map::LinearMap::with_capacity(start_capacity);
-        $( map.insert($key, $val); )*
+        $( map.insert($key.to_string(), $val.to_string()); )*
```

Comment:
Since we know that the LinearMap needs two strings calling to_string simplify the macro invocation. Not really a necessary change so I can remove this if wanted.

Reply:
Looks good.

Reply:
You are correct. This does causes redundant clones. I thought that to_string took a self, but it takes a &self. It is probably better to use .into() instead.